lib/pool-base.js

```diff
@@ -379,7 +379,7 @@ function PoolBase(options, processTask, createConnectionPool, pingPromise) {
     connectionInCreation = false;
     idleConnections.push(conn);
     pool.emit('idle-conn');
-    pool.emit('new-conn');
+    pool.emit('new-conn', conn);
```

Comment:
Just a small improvement: better to use async event emiting then:
`process.nextTick(() => pool.emit('new-conn', conn));` 
(and it will solve error in tests suite btw)